src/hotspot/os_cpu/linux_riscv/prefetch_linux_riscv.inline.hpp

```diff
@@ -27,12 +27,22 @@
 #define OS_CPU_LINUX_RISCV_VM_PREFETCH_LINUX_RISCV_INLINE_HPP
 
 #include "runtime/prefetch.hpp"
+#include "runtime/stubRoutines.hpp"
 
-
-inline void Prefetch::read (const void *loc, intx interval) {
+inline void Prefetch::read(const void *loc, intx interval) {
+    void (*stub)(const void*, intptr_t) =
+        (void (*)(const void*, intptr_t))StubRoutines::riscv::prefetch_r();
+    if (interval >= 0 && stub != NULL) {
+        stub(loc, interval);
```

Comment:
I am not sure if it really worth it to call a stub for read / write here. It looks to me not a big issue for the case the stub tries to catch and resolve. And I see aarch64 simply plant a 'prfm' instruction for prefetching [1]. I guess we might can do the same?

Reply:
We would need to check for `UseZicbop` in any case; the access to a global variable is then required.

It would be the same issue as https://github.com/openjdk/jdk/pull/10884/files/e968f7164124dcf560807c9ff7765e6f82b64cdd#diff-e3c18b8b83898e82b5a3069319df6a47468e91cc2527bf065e704a685a20f26bR5196 without the stub.

I've to admit that the `interval` naming here is confusing since no implementation ever uses it as an interval but alway as an offset. Also, the callers assume it to be an offset, like `ContiguousSpace::prepare_for_compaction` for example.

If we are to simply ignore the interval parameter, then we can also remove all uses of `PrefetchCopyIntervalInBytes`, `PrefetchScanIntervalInBytes`,

Reply:
Yes, I agree that a check for UseZicbop option would be necessary. But I still don't understand why we should implement this through a stub here. It looks to me that CPP code with inline assembly would also do. At least this could help eliminate the prologue & epilogue cost of calling the stub.

Reply:
We can definitely go with inline assembly here, just not sure that `prefetch.r` or `prefetch.w` would be recognized by the assembler. However, given that `prefetch.*` is an `ori` under the hood, we can simply use that.